libr/core/disasm.c

```diff
@@ -948,13 +948,36 @@ static void handle_show_functions(RCore *core, RDisasmState *ds) {
 				handle_print_offset (core, ds);
 				r_cons_printf ("     ");
 			}
-			r_cons_printf ("%s; %s %s %s %s@ %s%s0x%x%s\n",
-				COLOR (ds, color_other),
-				(var->kind == 'v') ? "var" : "arg",
-				var->type, var->name, spaces,
-				core->anal->reg->name[R_REG_NAME_BP],
-				(var->kind == 'v') ? "-" : "+",
-				var->delta, COLOR_RESET (ds));
+			r_cons_printf ("%s; ", COLOR (ds, color_other));
+			switch (var->kind) {
+			case 'a':
+				r_cons_printf ("arg %s %s @ %s+0x%x",
+					var->type, var->name,
+					core->anal->reg->name[R_REG_NAME_BP],
+					var->delta);
+				break;
+			case 'v':
+				r_cons_printf ("var %s %s @ %s-0x%x",
+					var->type, var->name,
+					core->anal->reg->name[R_REG_NAME_BP],
+					var->delta);
+				break;
+			case 'A':
+				if(var->delta==0){
+					r_cons_printf ("arg %s %s @ ecx",
+						var->type, var->name);
+				}else if(var->delta ==1){
+					r_cons_printf ("arg %s %s @ edx",
+						var->type, var->name);
+				}else{
```

Comment:
missing spaces around the `if` , `else`, `==`, parenthesis and so on :P

also i think it will be better to just use `switch(var->delta) { case 0: case 1: default }` instead of ifelse blocks